tests/e2e/vsan_test.go

```diff
@@ -60,44 +57,97 @@ func (s *VsanTestSuite) SetUpSuite(c *C) {
+	c.Assert(err, IsNil, Commentf(out))
+
+	invalidVsanOpts := [2]string{"-o " + adminclicon.VsanPolicyFlag + "=IDontExist", "-o " +
+		               adminclicon.VsanPolicyFlag + "=" + invalidContentPolicyName}
 	for _, option := range invalidVsanOpts {
 		invalidVolName := inputparams.GetVolumeNameWithTimeStamp("vsanVol") + "@" + s.vsanDSName
 		out, _ = dockercli.CreateVolumeWithOptions(s.config.DockerHosts[0], invalidVolName, option)
 		c.Assert(strings.HasPrefix(out, ErrorVolumeCreate), Equals, true)
 	}
 
-	misc.LogTestStart(volCreateTest, "TestVSANPolicy")
+	misc.LogTestEnd("", c.TestName())
+}
+
+// The purpose of this test is to verify:
+// 1) a volume can be created with vsan policy specified
+// 2) A vsan policy cannot be removed if volumes still use it
+
+// Test step:
+// 1. create a vsan policy
+// 2. run "vmdkops_admin policy ls", check the "Active" column of the output to make sure it
+// is shown as "Unused"
+// 3. create a volume on vsanDatastore with the vsan policy we created
+// 4. run "docker volume inspect" on the volume to verify the output "vsan-policy-name" field
+// 5. run "vmdkops_admin policy ls", check the "Active" column of the output to make sure it
+// is shown as "In use by 1 volumes"
+// 6. run "vmdkops_admin policy rm" to remove the policy, which should fail since the volume is still
+// use the vsan policy
+func (s *VsanTestSuite) TestDeleteVsanPolicyAlreadyInUse(c *C) {
+	misc.LogTestStart("", c.TestName())
+	s.volumeName = ""
+	out, err := admincli.CreatePolicy(s.config.EsxHost, adminclicon.PolicyName, adminclicon.PolicyContent)
+	c.Assert(err, IsNil, Commentf(out))
+
+	res := admincli.VerifyActiveFromVsanPolicyListOutput(s.config.EsxHost, adminclicon.PolicyName, "Unused")
+	c.Assert(res, Equals, true, Commentf("vsanPolicy should be \"Unused\""))
+
+	s.volumeName = inputparams.GetVolumeNameWithTimeStamp("vsanVol") + "@" + s.vsanDSName
+	vsanOpts := " -o " + adminclicon.VsanPolicyFlag + "=" + adminclicon.PolicyName
+	out, err = dockercli.CreateVolumeWithOptions(s.config.DockerHosts[0], s.volumeName, vsanOpts)
+	c.Assert(err, IsNil, Commentf(out))
+
+	policyName, err := verification.GetAssociatedPolicyName(s.config.DockerHosts[0], s.volumeName)
+	c.Assert(err, IsNil, Commentf("Get associated policy for volume %s failed", s.volumeName))
+	c.Assert(policyName, Equals, adminclicon.PolicyName, Commentf("The name of vsan policy used by volume "+s.vsanDSName+" returns incorrect value "+policyName))
+
+	res = admincli.VerifyActiveFromVsanPolicyListOutput(s.config.EsxHost, adminclicon.PolicyName, "In use by 1 volumes")
+	c.Assert(res, Equals, true, Commentf("vsanPolicy should be \"In use by 1 volumes\""))
+
+	out, err = admincli.RemovePolicy(s.config.EsxHost, adminclicon.PolicyName)
+	log.Printf("Remove vsanPolicy \"%s\" returns with %s", adminclicon.PolicyName, out)
+	c.Assert(out, Matches, "Error: Cannot remove.*", Commentf("vsanPolicy is still used by volumes and cannot be removed"))
+
```

Comment:
Need to add following verification after removing policy.

invoke `ListPolicy` (`admincli policy ls`) and make sure the policy `adminclicon.PolicyName` is not part of the output ... that validate the policy removal part.